apple_static_framework_import omits framework search paths for swiftinterface_imports and swift_module_imports

Hello, 

I am trying to build a framework A (`apple_framework`)  that depends on a pre-build static framework B (`apple_library`) that provides only `.swiftinterface` files. 

After updating to the latest release (3.2.1) i am getting the following error:

```
Compiling Swift module //<removed>:A_swift failed: (Exit 1): worker failed: error executing command (from target //<removed>:A_swift)
  (cd /private/var/tmp/_bazel_vakhid.betrakhmadov/5192802856cedd8189e7bb6f350ab59c/execroot/<removed> && \
  exec env - \
    APPLE_SDK_PLATFORM=iPhoneSimulator \
    APPLE_SDK_VERSION_OVERRIDE=17.2 \
    SWIFT_AVOID_WARNING_USING_OLD_DRIVER=1 \
    XCODE_VERSION_OVERRIDE=15.2.0.15C500b \
  bazel-out/darwin_arm64-opt-exec-2B5CBBC6-ST-c2e9848f5e1f/bin/external/build_bazel_rules_swift/tools/worker/worker swiftc @bazel-out/<removed>/A.swiftmodule-0.params)
# Configuration: b2a30624c41d574af473f1624cb44d84178037036a7956fb6286baac315349d9
# Execution platform: @local_config_platform//:host
<removed>/SomeSwiftFileInA.swift:9:8: error: no such module 'B'
import B
       ^
error: fatalError
Target //<removed>:A failed to build
```

Before updating i was on 2.3.0 and it worked fine.


What i found out so far is that `@bazel-out/<removed>/A.swiftmodule-0.params` is missing framework search path `-F<removed>/A.xcframework/ios-arm64_x86_64-simulator` for B and that this was prob. caused by a possible regression introduced by [this](https://github.com/bazelbuild/rules_apple/commit/043414543f93855beaf381d939843c60ad438c92#diff-554ab20e992ae7149038b68824662e27657acce05935ab52637f9bda1f8cfa28L184) commit in 3.1.0 by removing lines 184 and 191, as `_framework_search_paths` [here](https://github.com/bazelbuild/rules_apple/blob/6ef5ca4ca405ac59b44f525b5cf2d29076c13937/apple/internal/apple_framework_import.bzl#L431) was not updated to include `swiftinterface_imports` and `swift_module_imports`.

If i append `swiftinterface_imports` and `swift_module_imports` to `_framework_search_paths` argument [here](https://github.com/bazelbuild/rules_apple/blob/6ef5ca4ca405ac59b44f525b5cf2d29076c13937/apple/internal/apple_framework_import.bzl#L431) it works.

--- a/apple/internal/apple_framework_import.bzl
+++ b/apple/internal/apple_framework_import.bzl
@@ -428,7 +428,7 @@ def _apple_static_framework_import_impl(ctx):
             disabled_features = disabled_features,
             features = features,
             framework_includes = _framework_search_paths(
-                framework.header_imports,
+                framework.header_imports + framework.swift_interface_imports + framework.swift_module_imports,
             ),
             header_imports = framework.header_imports,
             kind = "static",

--- a/apple/internal/apple_framework_import.bzl
+++ b/apple/internal/apple_framework_import.bzl
@@ -428,7 +428,9 @@ def _apple_static_framework_import_impl(ctx):
             disabled_features = disabled_features,
             features = features,
             framework_includes = _framework_search_paths(
-                framework.header_imports + framework.swift_interface_imports + framework.swift_module_imports,
+                framework.header_imports +
+                framework.swift_interface_imports +
+                framework.swift_module_imports,
             ),
             header_imports = framework.header_imports,
             kind = "static",

